pdns/recursordist/test-syncres_cc4.cc

```diff
@@ -1022,7 +1022,7 @@ BOOST_AUTO_TEST_CASE(test_dnssec_bogus_rrsig_signed_with_unknown_dnskey)
   vector<DNSRecord> ret;
   int res = sr->beginResolve(target, QType(QType::NS), QClass::IN, ret);
   BOOST_CHECK_EQUAL(res, RCode::NoError);
-  BOOST_CHECK_EQUAL(sr->getValidationState(), vState::BogusNoValidRRSIG);
+  BOOST_CHECK_EQUAL(sr->getValidationState(), vState::BogusNoValidDNSKEY);
```

Comment:
That does not seem right to me, as in that case we have a valid DNSKEY corresponding to the DS but the RRSIG has been signed by a different key that is not present in the DNSKEY set.
rfc8914 defines the code 9 as "DNSKEY Missing: A DS record existed at a parent, but no supported matching DNSKEY record could be found for the child".

Reply:
Yes, It seems validateWithKeySet nees to become a bit smarter.